### PR TITLE
[FSDP] Remove the dependency of ``_symbolic_trace`` in ``wrap``

### PR DESCRIPTION
**Pull request by LinjianMa**
Same as #81339, this is used to fix internal tests where ``torch.fx`` is not available in the module and ``wrap`` of FSDP is imported. With this there should be no import errors for FSDP when ``torch.fx`` is not available.

### CODE REVIEW

File: torch/distributed/fsdp/wrap.py [hunk 1]
```diff
@@ -318,10 +317,13 @@ class ParamExecOrderWrapPolicy:
...
             within each transformer layer.
 
-        tracing_config (Optional[TracingConfig]):
-            The configuration used to perform symbolic tracing at FSDP constructor to get the module and
-            parameter execution order. If set as None, then symbolic tracing is not enabled, and one forward
-            as well as backward iteration are needed to get the parameter execution order.
+        tracing_config:
+            The configuration used to perform symbolic tracing at FSDP
+            constructor to get the module and parameter execution order. The
+            type of ``tracing_config`` needs to be either None or
```
Comment (awgu) [inline torch/distributed/fsdp/wrap.py:323]: nit: before this PR
```suggestion
            type of ``tracing_config`` needs to be either ``None`` or
```

File: torch/distributed/fsdp/wrap.py [hunk 2]
```diff
@@ -318,10 +317,13 @@ class ParamExecOrderWrapPolicy:
...
+            ``TracingConfig``. If set as None, then symbolic tracing is not
```
Comment (awgu) [inline torch/distributed/fsdp/wrap.py:324]: nit: 
```suggestion
            ``TracingConfig``. If set as ``None``, then symbolic tracing is not
```

File: torch/distributed/fsdp/wrap.py [hunk 3]
```diff
@@ -318,10 +317,13 @@ class ParamExecOrderWrapPolicy:
             wrapping each transformer layer into one FSDP unit, and can be easily combined with checkpointing
             within each transformer layer.
 
-        tracing_config (Optional[TracingConfig]):
-            The configuration used to perform symbolic tracing at FSDP constructor to get the module and
-            parameter execution order. If set as None, then symbolic tracing is not enabled, and one forward
-            as well as backward iteration are needed to get the parameter execution order.
+        tracing_config:
```
Comment (awgu) [inline torch/distributed/fsdp/wrap.py:320]: I think we can leave the type annotation in the docstring but omit it from the code. This way we at least still know what the type should be.

**Review by awgu — APPROVED** [review]
LGTM. Stamping to unblock but left minor comments.

**Review by zhaojuanmao — APPROVED** [review]

File: torch/distributed/fsdp/wrap.py (hunk 3 above)
Comment (LinjianMa) [reply torch/distributed/fsdp/wrap.py:320]: sounds good!

**Review by Neilblaze — APPROVED** [review]

**Pull request closed without merging**